### PR TITLE
API Final Cleanup

### PR DESCRIPTION
**Pull request by HankG**
This has the OpenID changes merged in from the API main branch plus all of the polishing made on the "final cleanup" branch which was originally based off the OpenID branch.

### CODE REVIEW

File: app/controllers/api/v1/posts_controller.rb [hunk 1]
```diff
@@ -65,20 +66,27 @@ def add_location_params(mapped_parameters)
...
         return unless params.has_key?(:photos)
+
         photo_guids = params[:photos]
         return if photo_guids.empty?
-        photo_ids = photo_guids.map {|guid| Photo.find_by!(guid: guid) }
-        raise InvalidArgument if photo_ids.length != photo_guids.length
-        mapped_parameters[:photos] = photo_ids
+
+        photos = photo_guids.map {|guid| Photo.find_by!(guid: guid) }
+                            .select {|p| p.author_id == current_user.person.id && p.pending }
```
Comment (frankrousseau) [inline app/controllers/api/v1/posts_controller.rb:74]: Is that possible to move this in a service?

File: app/controllers/api/v1/search_controller.rb [hunk 2]
```diff
@@ -46,6 +28,31 @@ def post_index
       def time_pager(query, query_time_field, data_time_field)
         Api::Paging::RestPaginatorBuilder.new(query, request).time_pager(params, query_time_field, data_time_field)
       end
+
+      def people_query
```
Comment (frankrousseau) [inline app/controllers/api/v1/search_controller.rb:32]: Is it possible to move this in a service?

File: spec/integration/api/posts_controller_spec.rb [hunk 3]
```diff
@@ -529,6 +540,18 @@
         )
         expect(response.status).to eq(403)
       end
+
+      it "fails without invalid token" do
```
Comment (frankrousseau) [inline spec/integration/api/posts_controller_spec.rb:544]: It should be with instead of without.

**Review by frankrousseau — COMMENTED** [review]
It looks good to me. As usual, I mentioned a few things in the review but they are only minor issues.

File: app/controllers/api/v1/posts_controller.rb (hunk 1 above)
Comment (HankG) [reply app/controllers/api/v1/posts_controller.rb:74]: Moved to service in latest commit

File: app/controllers/api/v1/search_controller.rb (hunk 2 above)
Comment (HankG) [reply app/controllers/api/v1/search_controller.rb:32]: This is pretty specific to the search controller except for the direct calls into the Person objects which are already abstracted out.  I think moving it to a service won't get us much.

File: spec/integration/api/posts_controller_spec.rb (hunk 3 above)
Comment (HankG) [reply spec/integration/api/posts_controller_spec.rb:544]: Fixed in latest commit

**Comment by HankG** [comment]
I'm going to squash and take off the WIP so can be integrated into main branch.

**Comment by HankG** [comment]
Ready for final merge.

File: app/controllers/api/v1/search_controller.rb (hunk 2 above)
Comment (frankrousseau) [reply app/controllers/api/v1/search_controller.rb:32]: Ok I prefer to see that querying complexity in services but it's just a matter of taste.

**Comment by frankrousseau** [comment]
Done! It's great too to see that a full working code for the Diaspora API is now available! That's a lot of work. I hope it will make its way to the official repository. It's a long awaited feature, the community will surely appreciate it!
By the way, thank you a lot for letting me review your code. I learned a lot through that experience.

**Pull request merged**